Modules/_abc.c

```diff
@@ -288,11 +288,6 @@ _exit_iter(_guarded_set *gs)
         PyObject *ref = PyList_GET_ITEM(gs->pending, pos);
         Py_INCREF(Py_None);
         PyList_SET_ITEM(gs->pending, pos, Py_None);
-
-        if (ref == Py_None) {
-            Py_DECREF(ref);
-            continue;
-        }
         if (PySet_Discard(gs->data, ref) < 0) {
```

Comment:
Oh, I'm sorry that I didn't leave comments here.
This is very tricky part.

Think about if PySet_Discard() failed.
Since we didn't pop from the list, None can be remained in the pending list.

When _exit_iter() called is next time, we must skip calling PySet_Discard() for
at least objects we successfully disgarded already.  That's why I set None
to the list temporary.  So if we get None from the list, we must skip it.

Otherwise, `PySet_Discard(gs->data, None)` always fail and we can't clear
the pending list eternally.

Reply:
Just to clarify, `set().discard(None)` doesn't fail, if an element is not in the set it is just a no-op. Or what you mean is more subtle?

Reply:
Oh, I thought it raise KeyError.  Sorry.
So we don't need to skip explicitly for such rare case.